bluefog/common/global_state.h

```diff
@@ -58,6 +60,15 @@ struct BluefogGlobalState {
   // Flag indicating whether timeline enabled.
   bool timeline_enabled = false;
 
+  // Background thread cycle time in milliseconds.  Fractional numbers are permitted.
+  double cycle_time_ms = 3;
+
+  // Time point when last cycle started.
+  std::chrono::steady_clock::time_point last_cycle_start;
```

Comment:
add unit

Reply:
Done

Reply:
time_point is a class instead of a simple wrap on the int. It does not need a unit actually. (Like duration you can compare it with second based duration and millisecond based duration).

template <class _Clock, class _Duration = typename _Clock::duration>
class _LIBCPP_TEMPLATE_VIS time_point
{
    static_assert(__is_duration<_Duration>::value,
                  "Second template parameter of time_point must be a std::chrono::duration");
public:
    typedef _Clock                    clock;
    typedef _Duration                 duration;
    typedef typename duration::rep    rep;
    typedef typename duration::period period;